tests/test_ur_driver.cpp

```diff
@@ -379,6 +379,30 @@ TEST_F(UrDriverTest, reset_rtde_client)
   ASSERT_EQ(g_ur_driver_->getControlFrequency(), target_frequency);
 }
 
+TEST_F(UrDriverTest, read_error_code)
+{
+  g_ur_driver_->startErrorCodeClientCommunication();
+  std::stringstream cmd;
+  cmd << "sec setup():" << std::endl << " protective_stop()" << std::endl << "end";
+  EXPECT_TRUE(g_ur_driver_->sendScript(cmd.str()));
+
+  auto error_codes = g_ur_driver_->getErrorCodes();
+  while (error_codes.size() == 0)
+  {
+    error_codes = g_ur_driver_->getErrorCodes();
+  }
+
+  ASSERT_EQ(error_codes.size(), 1);
+  ASSERT_EQ(error_codes.at(0).message_code, 209);
```

Comment:
Maybe having error codes as an enum somewhere would be nice so once could understand error codes when reading the code :-)

Reply:
> Maybe having error codes as an enum somewhere would be nice so once could understand error codes when reading the code :-)

I will suggest a link to the documentation of the error codes instead, to avoid maintain the list in this repo.
Currently all the manuals, including an error code description, can be accessed from https://myur.universal-robots.com/manuals